src/session.cpp

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include <src/session.hpp>
+
+namespace obmc
+{
+namespace session
+{
+void SessionItem::delete_()
+{
+    auto manager = managerWeakPtr.lock();
+    if (manager)
+    {
+        SessionManager::SessionIdentifier sessionId =
+            SessionManager::parseSessionId(this->sessionID());
+        if (!manager->remove(sessionId)) {
+            throw InternalFailure();
+        }
+    }
+}
+
+void SessionItem::resetCleanupFn(SessionManager::SessionCleanupFn&& cleanup)
+{
+    this->cleanupFn.reset();
+    this->cleanupFn.emplace(cleanup);
+}
+
+void SessionItem::adjustSessionOwner(uid_t)
+{
+    // Define interface to adjust session owner by user ID.
+    // Currently, not implemented.
```

Comment:
Why have this stub at all? If it is ever needed, we can add it then.

Reply:
Currently, each `not implemented` method is not required for the next RC, but these must-haves for the feature of the control session by admin role from UI.  I prefer to define the public interface of the shared library to be considered by all library consumers. This is only for saving time, but I can implement those methods for the initial version either.

Reply:
Fixed